tests/src/rules/no-duplicates.js

```diff
@@ -428,99 +428,225 @@ context('TypeScript', function () {
+            import type {} from './module';
+            import {} from './module2';
+          `,
+          ...parserConfig,
+        }),
+        test({
+          code: `
+            import type { Identifier } from 'module';
+
+            declare module 'module2' {
+              import type { Identifier } from 'module';
+            }
+
+            declare module 'module3' {
               import type { Identifier } from 'module';
+            }
+          `,
+          ...parserConfig,
+        }),
+      ].concat(!tsVersionSatisfies('>= 4.5') || !typescriptEslintParserSatisfies('>= 5.7.0') ? [] : [
+        // #2470: ignore duplicate if is a typescript inline type import
+        test({
+          code: "import { type x } from './foo'; import y from './foo'",
+          ...parserConfig,
+        }),
+        test({
+          code: "import { type x } from './foo'; import { y } from './foo'",
+          ...parserConfig,
+        }),
+        test({
+          code: "import { type x } from './foo'; import type y from 'foo'",
+          ...parserConfig,
+        }),
+      ]);
+
+      const invalid = [
+        test({
+          code: "import type x from './foo'; import type y from './foo'",
+          output: "import type x from './foo'; import type y from './foo'",
+          ...parserConfig,
+          errors: [
+            {
+              line: 1,
+              column: 20,
+              message: "'./foo' imported multiple times.",
+            },
+            {
+              line: 1,
+              column: 48,
+              message: "'./foo' imported multiple times.",
+            },
+          ],
+        }),
+        test({
+          code: "import type x from './foo'; import type x from './foo'",
+          output: "import type x from './foo'; ",
+          ...parserConfig,
+          errors: [
+            {
+              line: 1,
+              column: 20,
+              message: "'./foo' imported multiple times.",
+            },
+            {
+              line: 1,
+              column: 48,
+              message: "'./foo' imported multiple times.",
+            },
+          ],
+        }),
+        test({
+          code: "import type {x} from './foo'; import type {y} from './foo'",
+          ...parserConfig,
+          output: `import type {x,y} from './foo'; `,
+          errors: [
+            {
+              line: 1,
+              column: 22,
+              message: "'./foo' imported multiple times.",
+            },
+            {
+              line: 1,
+              column: 52,
+              message: "'./foo' imported multiple times.",
+            },
+          ],
+        }),
+      ].concat(!tsVersionSatisfies('>= 4.5') || !typescriptEslintParserSatisfies('>= 5.7.0') ? [] : [
+        test({
+          code: "import {type x} from './foo'; import type {y} from './foo'",
+          ...parserConfig,
+          options: [{ 'prefer-inline': false }],
+          output: `import {type x,y} from './foo'; `,
```

Comment:
I want to point out this test.  no-duplicates will collapse these imports like this even before this PR.  Is there a different way we would want to handle this case?  When combined with this [PR](https://github.com/typescript-eslint/typescript-eslint/pull/5050), it would convert `y` to `type y`.  @ljharb

Reply:
Is this a difference in Flow vs TS?

Reply:
I believe so, but I don't think I fully grok the question as well 😄 so sorry if not a direct answer.  From my very limited knowledge, Flow doesn't have inline type imports.  So this test is a bit obtuse and really just shows that for someone who wrote their imports this way, they still have a path (with other rules) to correct the imports to proper inline types.

Reply:
I thought Flow used to be the only thing that has inline type imports, and TS only recently added them?

Reply:
Oh!  I'm not familiar with Flow but did some looking.  I'm only seeing top level type imports but I didn't spend a lot of time looking.  And yep, TS added them in 4.5 so this is the last PR in a [bucket](https://github.com/import-js/eslint-plugin-import/pull/2475#issuecomment-1304888131) of 3 PRs to allow codebases to migrate from top level (separate from value imports) type imports to inline type imports.

Reply:
If there’s any autofix that would conflict based on using TS or Flow, then for those we need to use suggestions instead.

Reply:
One thing to note is that this will only happen if TS > 4.5.  A Flow project would presumably fail upstream if someone tried `import { type x }`.  So I think this test case is not applicable to Flow

Reply:
@bradzacher Do you have any additional pieces of info to add here?

Reply:
@snewcomer meaning, the rule detect TS 4.5+ and only does this autofix in that case?

Reply:
More-so you won't even be able to build the project if you use invalid syntax.  In this case below, this is a project TS < 4.5.  I would assume similar things would happen in Flow if using inline types (assuming they don't have support for them at all)

<img width="489" alt="Screen Shot 2022-11-25 at 7 17 23 AM" src="https://user-images.githubusercontent.com/7374640/203994415-7392236c-045b-41a5-9adb-ffd9bd4cb6f2.png">

Reply:
Right - and an eslint autofix should never cause a project to become unbuildable or change semantics. That's what "safe" means in the context of autofixing.

Reply:
So to clarify before I investigate an addition here - I need to prevent the case where somebody erroneously sets ‘prefer-inline: true’ but their TS version (or Flow) doesn’t support inlining types?

Reply:
> erroneously sets ‘prefer-inline: true’

To clarify further, meaning they opt-into something that their project does not support.

Reply:
I'm sorry to bother.  Always waffling whether to bother busy maintainers like yourself.  Do you happen to have a thought here?

> So to clarify before I investigate a new commit to this PR - I need to prevent the case where somebody erroneously sets ‘prefer-inline: true’ but their TS version (or Flow) doesn’t support inlining types?

Reply:
Thank you @ljharb!  How does [this](https://github.com/import-js/eslint-plugin-import/pull/2475/commits/f495fd931c8b48d300f5ce9d3c7274f2fffb35b6) look?

Reply:
Looks great!